### PR TITLE
NeoForge support

### LINKED ISSUE #2089: Support NeoForge
### Minecraft Development for IntelliJ plugin version

2023.1-1.6.8

### Description of the feature request

Please add support for https://neoforged.net/

### PR DESCRIPTION
**Pull request by RedNesto**
Fixes #2089

### CODE REVIEW

**Comment by Earthcomputer** [comment]
I tried selecting use parchment but the combo box is still grayed out:
image

**Comment by RedNesto** [comment]
You probably have selected MC 1.20.4, which doesn't have parchments mappings yet.
I should add an error label in that case.

**Comment by Earthcomputer** [comment]
It's fine to use Parchment for an older Minecraft version, so you could include them in the list with a warning instead.

File: src/main/kotlin/platform/neoforge/creator/asset-steps.kt [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        val license = data.getUserData(LicenseStep.KEY) ?: return
+        val description = data.getUserData(DescriptionStep.KEY) ?: ""
+        val updateUrl = data.getUserData(UpdateUrlStep.KEY) ?: ""
+        val authors = data.getUserData(AuthorsStep.KEY) ?: emptyList()
+        val website = data.getUserData(WebsiteStep.KEY) ?: ""
+        val useMixins = data.getUserData(UseMixinsStep.KEY) ?: false
+
+        val nextMcVersion = when (val part = mcVersion.parts.getOrNull(1)) {
+            // Mimics the code used to get the next Minecraft version in NeoForge's MDK
+            // https://github.com/MinecraftForge/MinecraftForge/blob/0ff8a596fc1ef33d4070be89dd5cb4851f93f731/build.gradle#L884
```
Comment (Earthcomputer) [inline src/main/kotlin/platform/neoforge/creator/asset-steps.kt]: This still links the MinecraftForge repo

File: src/main/kotlin/platform/neoforge/version/NeoForgeVersion.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                    parts.add(0, SemanticVersion.Companion.VersionPart.ReleasePart(1, "1"))
+                    SemanticVersion(parts)
+                } else null
+            }.distinct()
+            .sortedDescending()
+            .toList()
+        return@lazy version
+    }
+
+    fun getForgeVersions(mcVersion: SemanticVersion): List<SemanticVersion> {
```
Comment (Earthcomputer) [inline src/main/kotlin/platform/neoforge/version/NeoForgeVersion.kt]: `getNeoForgeVersions`?

**Review by Earthcomputer — APPROVED** [review]

**Comment by WitherTM** [comment]
> You probably have selected MC 1.20.4, which doesn't have parchments mappings yet. I should add an error label in that case.

Sorry to reply to a 2-week-old post, but the 1.20.2 parchment mappings perfectly work on 1.20.4 with NeoForge

**Comment by RedNesto** [comment]
This has already been handled in https://github.com/minecraft-dev/MinecraftDev/pull/2196/commits/331bf2b15ee69537da60425db5c63977334ecd59

**Comment by WitherTM** [comment]
> This has already been handled in [331bf2b](https://github.com/minecraft-dev/MinecraftDev/commit/331bf2b15ee69537da60425db5c63977334ecd59)

thx, I should've looked :)

**Pull request merged**